stan/math/rev/fun/log_determinant.hpp

```diff
@@ -4,33 +4,32 @@
 #include <stan/math/rev/meta.hpp>
 #include <stan/math/rev/core.hpp>
 #include <stan/math/rev/fun/typedefs.hpp>
+#include <stan/math/rev/functor/reverse_pass_callback.hpp>
+#include <stan/math/rev/core/arena_matrix.hpp>
 #include <stan/math/prim/err.hpp>
 #include <stan/math/prim/fun/Eigen.hpp>
 #include <stan/math/prim/fun/typedefs.hpp>
 
 namespace stan {
 namespace math {
 
-template <typename EigMat, require_eigen_vt<is_var, EigMat>* = nullptr>
-inline var log_determinant(const EigMat& m) {
-  using Eigen::Matrix;
-
-  math::check_square("log_determinant", "m", m);
-
-  Eigen::FullPivHouseholderQR<promote_scalar_t<double, EigMat>> hh
-      = m.val().fullPivHouseholderQr();
-
-  vari** varis
-      = ChainableStack::instance_->memalloc_.alloc_array<vari*>(m.size());
-  Eigen::Map<matrix_vi>(varis, m.rows(), m.cols()) = m.vi();
-
-  double* gradients
-      = ChainableStack::instance_->memalloc_.alloc_array<double>(m.size());
-  Eigen::Map<matrix_d>(gradients, m.rows(), m.cols())
-      = hh.inverse().transpose();
-
-  return var(new precomputed_gradients_vari(hh.logAbsDeterminant(), m.size(),
-                                            varis, gradients));
+template <typename T, require_rev_matrix_t<T>* = nullptr>
+inline var log_determinant(const T& m) {
```

Comment:
Use the implementation from here: https://github.com/stan-dev/math/blob/654aa5119a3d7870c527cb9f3cedb487b72474e3/stan/math/rev/fun/log_determinant.hpp#L26

I think it just changes 'fullPivHouseholderQr` to `colPivHouseholderQr` so you could just do that too.

Reply:
Done